cmd/ci-operator-prowgen/main.go

```diff
@@ -607,8 +617,11 @@ func generateJobs(
 
 	for _, element := range configSpec.Tests {
 		var podSpec *kubeapi.PodSpec
+		if element.Secret != nil {
+			element.Secrets = append(element.Secrets, element.Secret)
+		}
```

Comment:
This does not look correct. We move `Secret` to `Secrets` here for each test, but then we do it again in `generatePodSpecRandom` and `generatePodSpecTemplate`? This should be only done once.

Reply:
I agree. I took a sort of "look for any reference of singular-Secret and handle it" in case `generatePodSpecRandom` and `generatePodSpecTemplate` were called elsewhere, but AFAICT they aren't. I'll clean this up next.